cmd_migrate_db.go

```diff
@@ -0,0 +1,804 @@
+		if err == nil {
+			logger.Infof("Skipping DB with prefix %s because the "+
+				"destination DB was marked as already having "+
+				"been the target of a successful migration. "+
+				"Tag reads: %s", prefix, marker)
+
+			continue
+		}
+		if err != channeldb.ErrMarkerNotPresent {
+			return err
+		}
+
+		// Create a separate meta db for each db to store the
+		// migration/verification state. This db will be deleted
+		// after the migration is successful.
+		metaDBPath := filepath.Join(
+			x.Source.Bolt.DataDir, prefix+"-migration-meta.db",
+		)
+		if x.ForceNewMigration {
+			fileExists := func(path string) bool {
+				_, err := os.Stat(path)
+				return !os.IsNotExist(err)
+			}
+
+			if fileExists(metaDBPath) {
+				os.Remove(metaDBPath)
+			}
+		}
+
+		// Open the db where we store the migration/verification state.
+		metaDB, err := bbolt.Open(metaDBPath, 0600, nil)
+		if err != nil {
+			logger.Errorf("Error opening db: %v", err)
+		}
+		defer metaDB.Close()
+
+		logger.Infof("Opened meta db at path: %s", metaDBPath)
+
+		// Configure and run migration.
+		cfg := migratekvdb.Config{
+			Logger:    logger.SubSystem("MIGKV"),
+			ChunkSize: x.ChunkSize,
+			MetaDB:    metaDB,
+		}
+
+		migrator, err := migratekvdb.New(cfg)
+		if err != nil {
+			return err
+		}
+
+		err = migrator.Migrate(context.Background(), srcDb, destDb)
+		if err != nil {
+			return err
+		}
+		logger.Info("Migration completed")
+
+		// We migrated the DB successfully, now we verify the migration.
+		err = migrator.VerifyMigration(
+			context.Background(), srcDb, destDb,
+		)
+		if err != nil {
+			return err
+		}
+
+		logger.Info("Verification completed")
+
+		// Migrate wallet created marker. This is done after the
+		// migration to ensure the verification of the migration
+		// succeeds.
+		//
+		// NOTE: We always need to add the wallet marker if the db is
+		// not a `bolt` db, which is already resticted by the
+		// destination db config.
+		if prefix == lncfg.NSWalletDB {
+			err := createWalletMarker(destDb, logger)
+			if err != nil {
+				return err
+			}
+		}
+
+		// If we get here, we've successfully migrated the DB and can
+		// now set the tombstone marker on the source database and the
+		// already migrated marker on the target database.
+		// We need to reopen the db in write mode.
+		srcDb.Close()
+		logger.Info("We are now opening the source db in write mode " +
+			"to set the tombstone marker. This might take a while " +
+			"(~10 minutes for large databases) to sync the " +
+			"freelist..., so please be patient it is the final " +
+			"step.")
+
+		srcDb, err = openSourceDb(x.Source, prefix, x.Network, false)
+		if err != nil {
+			return err
+		}
+
+		if err := addMarker(srcDb, channeldb.TombstoneKey); err != nil {
+			return err
+		}
+
+		// Create migration completed file, this will only create the
+		// file for bolt databases.
+		if err := createMigrationCompletedFile(x.Source, prefix,
```

Comment:
nit: i think ideally this file should be added right after both db's are closed.